examples/globe_wfs_extruded.js

```diff
@@ -0,0 +1,122 @@
+/* global itowns, document, renderer */
+// # Simple Globe viewer
+
+// Define initial camera position
+var positionOnGlobe = { longitude: 4.818, latitude: 45.7354, altitude: 3000 };
+var promises = [];
+
+// `viewerDiv` will contain iTowns' rendering area (`<canvas>`)
+var viewerDiv = document.getElementById('viewerDiv');
+
+// Instanciate iTowns GlobeView*
+var globeView = new itowns.GlobeView(viewerDiv, positionOnGlobe, { renderer: renderer });
+function addLayerCb(layer) {
+    return globeView.addLayer(layer);
+}
+
+// Define projection that we will use (taken from https://epsg.io/3946, Proj4js section)
+itowns.proj4.defs('EPSG:3946',
+    '+proj=lcc +lat_1=45.25 +lat_2=46.75 +lat_0=46 +lon_0=3 +x_0=1700000 +y_0=5200000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs');
+
+// Add one imagery layer to the scene
+// This layer is defined in a json file but it could be defined as a plain js
+// object. See Layer* for more info.
+promises.push(itowns.Fetcher.json('./layers/JSONLayers/Ortho.json').then(addLayerCb));
+
+// Add two elevation layers.
+// These will deform iTowns globe geometry to represent terrain elevation.
+promises.push(itowns.Fetcher.json('./layers/JSONLayers/WORLD_DTM.json').then(addLayerCb));
+promises.push(itowns.Fetcher.json('./layers/JSONLayers/IGN_MNT_HIGHRES.json').then(addLayerCb));
+
+function colorFunctionLine(layer, node, featureCollection) {
+    var i;
+    var featureProperties;
+    var rgb;
+    var colors = [];
+
+    for (i = 0; i < featureCollection.features.length; i++) {
+        featureProperties = featureCollection.features[i].properties;
+
+        rgb = featureProperties.couleur.split(' ');
+        colors.push(new itowns.THREE.Color(rgb[0] / 255, rgb[1] / 255, rgb[2] / 255));
+    }
+
+    itowns.FeatureProcessing.assignColorsToFeatureCollection(
+        featureCollection, featureCollection.children[0], colors);
+
+    featureCollection.children[0].material.linewidth = 5;
+}
+
+globeView.addLayer({
+    update: itowns.FeatureProcessing.update(colorFunctionLine),
```

Comment:
`colorFunctionLine` is used once when creating the mesh so `colorFunctionLine` should be an argument of `convertAndExtrude`.

Reply:
This code come from example/wfs.js. 
Creating the mesh and coloring the mesh is separated in the code.
I don't want to refactor / re design this part of iTowns now.

Reply:
This `colorFunctionLine` stuff was written under this simple assumption: the geojson -> mesh feature wouldn't not be customisable, except for the color part.

Now you're breaking this assumption: this is a good thing, but you can't avoid refactoring the code that now doesn't make sense.

The color selection code is only used at mesh creation time which is now the responsability of your `convert` function: you should adapt the old code to the new feature.

Reply:
I understand that, it's a good idea, but for me, it's a big work to refactor that.
Can you be more precise on the refactor you want ?

Reply:
Move the mesh creation dependencies from `update` to `convert`.
So your `convert` function could receive an object with properties allowing the user to select the value of these attributes:
- color: could be either a THREE.Color or a function or undefined (= you pick a random color for a given layer)
- altitude: could be a constant, a function or undefined (= 0)
- extrude: could be a constant, a function or undefined (= no extrusion)

I don't expect this code shuffling to be a big refacto and it should greatly improve the API.

Reply:
I 've update the code.

Now the color function is very easy to read and write !!

But now, the material is not accessible for the user. 
For example, for the Lyon bus lines, I had to put the material change in Feature2Mesh.
https://github.com/iTowns/itowns/pull/519/files#diff-2a78502be7a11a150dcce98113215193R145

So, I plan to add another callback to update the material (I'll do it tomorrow).
Good idea ? @peppsac

Reply:
For tiles and points we already have a callback: https://github.com/iTowns/itowns/blob/master/src/Process/TiledNodeProcessing.js#L47 and https://github.com/iTowns/itowns/blob/master/src/Process/PointCloudProcessing.js#L236 so it would make sense to add the same thing for FeatureProcessing.

On the other hand, you can also add another argument to your convert function: `material` that could be either:
- a `Material` (like we do for 3dtiles: https://github.com/iTowns/itowns/blob/master/src/Core/Scheduler/Providers/3dTiles_Provider.js#L170)
- or a collection of material property names, which could either constant or functions. For instance:
```js
convert({ material: new THREE.LineDashedMaterial() });
// or
convert({ material: { linewidth: 5, linecap: myLineCapFunction } });
```
WDYT?